katago_server/runs/models/run.py

```diff
@@ -46,7 +46,9 @@ class RunStatus(TextChoices):
     id = AutoField(primary_key=True)
     created_at = DateTimeField(_("creation date"), auto_now_add=True, db_index=True)
     status = CharField(_("run status"), max_length=15, choices=RunStatus.choices, db_index=True, default=RunStatus.INACTIVE,)
-    # TODO enforce that name is UNIQUE
+    # TODO is it possible to make this the primary key?
```

Comment:
The proper way to do this is:

- in the viewset

```diff
 class RunViewSet(viewsets.ModelViewSet):
     """
     API endpoint that allows runs to be viewed or edited.
     """
     queryset = Run.objects.all()
     permission_classes = [IsAdminUser | ReadOnly]
     serializer_class = RunSerializer
     filterset_fields = ["status"]
+    lookup_filed = 'name'
```

and not to break the automatically generated url, in each serializers

```diff
 class RunSerializerForClient(HyperlinkedModelSerializer):
     """
     Serializer exposing only the fields of a run that a self-play client needs on startup.
     """
 
     class Meta:
         model = Run
         fields = [
             "id",
             "url",
             "name",
             "data_board_len",
             "inputs_version",
             "max_search_threads_allowed",
         ]
+       extra_kwargs = {
+            'url': {'lookup_field': 'licensenumber'}
+        }
```

Last important point (but already done here) is to have `db_index=True` in the model so the query remains efficient.

Reply:
It should be "name" instead of "licensenumber", right?

Reply:
yes, copy pasta from the doc :/